### PR TITLE
elm-format: 0.8.4 -> 0.8.5

### PR DESCRIPTION
**Pull request by avh4**
###### Motivation for this change

New release of elm-format.

Notes:
- elm-format is now split into a few internal packages, which are now added via cabal2nix
- bimap-0.3.3 is added temporarily.  I think 0.4.0 should be safe to switch to (though haven't tried it yet), but I'd have to cut a new release tag to relax the version bounds.
- elm-format now uses ghc-8.10.4; I switched pkgs/development/compilers/elm/default.nix to use 8.10.3 instead of 8.8.3.  I tried building elm, elmi-to-json, and elm-instrument and they all seemed to work fine, but I think elm-format would also probably compile okay with 8.8.3 (though I didn't check for sure).   I think ideally things should be set up to allow the different tools to have different ghc versions, esp as I'm planning to move elm-format to ghc-9 in the next release, but I'm not sure I know enough about nixpkgs to attempt a change like that.

cc @turboMaCk  @domenkozar  

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by turboMaCk** [comment]
@avh4 Don't you want to add yourself to the maintainers as well? I think it would be good to have a project author listed as maintainer.

You can add yourself to this file (me as an example) https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix#L10433-L10438 and than to the package meta.

- this is usually its own commit so no need to squash it together with package update
- [Instructions](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix#L24) to get the github id

File: pkgs/development/compilers/elm/packages/elm-format.nix [hunk 1]
```diff
@@ -1,40 +1,40 @@
...
-, bytestring, cmark, containers, directory, filepath, free, HUnit
-, indents, json, mtl, optparse-applicative, parsec, process
-, QuickCheck, quickcheck-io, split, lib, tasty, tasty-golden
-, tasty-hunit, tasty-quickcheck, text
+{ mkDerivation, ansi-wl-pprint, avh4-lib, base, bimap, cmark
+, containers, elm-format-lib, elm-format-test-lib, fetchgit, json
+, lib, mtl, optparse-applicative, parsec, QuickCheck, quickcheck-io
+, relude, tasty, tasty-hspec, tasty-hunit, tasty-quickcheck, text
 }:
 mkDerivation {
```
Comment (turboMaCk) [inline pkgs/development/compilers/elm/packages/elm-format.nix:6]: we can use `rec` here for recursive...

```suggestion
mkDerivation rec {
```

File: pkgs/development/compilers/elm/packages/elm-format.nix [hunk 2]
```diff
@@ -1,40 +1,40 @@
...
   homepage = "https://elm-lang.org";
   description = "A source code formatter for Elm";
   license = lib.licenses.bsd3;
+  postPatch = ''
+    mkdir -p ./generated
+    cat <<EOHS > ./generated/Build_elm_format.hs
+    module Build_elm_format where
+
+    gitDescribe :: String
+    gitDescribe = "0.8.5"
```
Comment (turboMaCk) [inline pkgs/development/compilers/elm/packages/elm-format.nix:37]: and then (with `rec`) refer to `version` attribute here:

```suggestion
    gitDescribe = ${version}
```

File: pkgs/development/compilers/elm/packages/avh4-lib.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+mkDerivation {
+  pname = "avh4-lib";
+  version = "0.0.0.1";
+  src = fetchgit {
+    url = "https://github.com/avh4/elm-format";
+    sha256 = "0bcjkcs1dy1csz0mpk7d4b5wf93fsj9p86x8fp42mb0pipdd0bh6";
+    rev = "80f15d85ee71e1663c9b53903f2b5b2aa444a3be";
+    fetchSubmodules = true;
+  };
+  postUnpack = "sourceRoot+=/avh4-lib; echo source root reset to $sourceRoot";
```
Comment (turboMaCk) [inline pkgs/development/compilers/elm/packages/avh4-lib.nix:15]: What is this doing?

**Review by turboMaCk — APPROVED** [review]
I've tested build of elm, elm-format, elm-instrument and elmi-to-json on Darwin and NixOS and all works.

File: pkgs/development/compilers/elm/packages/avh4-lib.nix (hunk 3 above)
Comment (avh4) [reply pkgs/development/compilers/elm/packages/avh4-lib.nix:15]: This is building a haskell library that's in a folder within the git repo.  This whole file was generated via `cabal2nix --no-haddock https://github.com/avh4/elm-format --revision "0.8.5" --subpath avh4-lib`

Comment (turboMaCk) [reply pkgs/development/compilers/elm/packages/avh4-lib.nix:15]: Ah thanks. I see that repository is now sort of monorepo containing few individual libraries :+1: Makes sense.

**Pull request merged**